src/nupic/algorithms/SDRClassifier.hpp

```diff
@@ -45,9 +44,9 @@ typedef cla_classifier::ClassifierResult ClassifierResult;
 
 namespace sdr_classifier {
 
-const UInt sdrClassifierVersion = 1;
+const UInt sdrClassifierVersion = 2;
 
-typedef Dense<UInt, Real64> Matrix;
+typedef std::map<UInt, std::map<UInt, Real64>> Matrix; //Matrix[r][c] = 0.0d
```

Comment:
I'm looking for comments on implementation of this 2D matrix. It's a mapping from 2d [UInt][UInt] -> Real64; 
- used to be DenseMatrix
- now considering: 
  - `map<int, map<int, double>>`
  - Eigen: `Matrix<Real64, Dynamic, Dynamic>`
  - plain Real64[][] //we sometimes have to grow it
  - vector<vector<Real>>

I'm inclining to eigen::Matrix as we need here concept of rows, cols.

Reply:
for now implemented using native `map<int, map<int, double>> map2d;` 
Eigen would be optional improvement for the future, if needed.

Reply:
I like this matrix type.  I prefer the standard library types because they're understood by everyone.

Reply:
I agree std:: classes are always good but I don't have any experience with eigen so I cannot say which is better.